internal/controllers/user_integrations_controller.go

```diff
@@ -1810,6 +1811,24 @@ func (udc *UserDevicesController) registerDeviceTesla(c *fiber.Ctx, logger *zero
 		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request body.")
 	}
 
+	if reqBody.TeslaAPIVersion == 0 {
+		reqBody.TeslaAPIVersion = 1
+	}
+
+	if reqBody.ExternalID == "" {
+		return fiber.NewError(fiber.StatusBadRequest, "Missing externalID parameter")
+	}
+
+	if reqBody.TeslaAPIVersion == 2 { // If version is 2, we are using fleet api which, which has token stored in cache
```

Comment:
To avoid magic numbers, we can add constants to the file. You don't have to use these names they are just an example.
``` golang
const (
teslaAPIv1 = 1
teslaAPIv2Fleet = 2
)